### PR TITLE
Refactor check.py so that groups of tests can be split into separate …

### PR DESCRIPTION
**Pull request by jgravelle-google**
…files

- Move helper util functions into test/shared.py
- Move scripts/support.py to test/support.py
- Split s2wasm tests into its own file

This is pretty half-baked and lazily done; there's probably better directory structure, `from shared import *` should probably get cleaned up, argument parsing in an import is weird, etc.
But, this is mostly a proof-of-concept and/or incremental improvement.

Benefits:
- Can run a single test suite without having to add command-line flags and more conditionals (main reason I did this)
- Better organization of test code. It's easier to navigate when files are smaller and more directed.

Thoughts/suggestions/concerns?

### CODE REVIEW

**Comment by kripken** [comment]
Overall this looks like a nice improvement. A few notes:

 * python conventions are `test_s2wasm`.
 * flake8 looks in `scripts/`, so by moving  code out we reduce coverage. might want to point it to the new locations too.
 * I don't follow "Can run a single test suite without having to add command-line flags and more conditionals" - what does the new command look like? do you mean you invoke the sub-script directly?

**Comment by jgravelle-google** [comment]
Correct, instead of `./check.py --suite s2wasm` or something, I run `test/s2wasm.py`.
The nice thing there is we don't need to modify `check.py` to check `if not options.test_suite or options.test_suite == 's2wasm':` before each test.

File: check.py [hunk 1]
```diff
@@ -14,296 +14,19 @@
...
+from test.support import run_command, split_wast
+from test.shared import *
+
+from test.s2wasm import s2wasmTests, linkerTests
 
 if options.interpreter:
   print '[ using wasm interpreter at "%s" ]' % options.interpreter
   assert os.path.exists(options.interpreter), 'interpreter not found'
 
 num_failures = 0
```
Comment (jgravelle-google) [inline check.py]: This also needs to be moved to `test/shared.py`, or it'll cause errors when there are failures.

File: check.py [hunk 2]
```diff
@@ -14,297 +14,21 @@
...
-options = parser.parse_args()
-requested = options.positional_args
+import json
+import os
+import shutil
+import subprocess
+import sys
+
+from scripts.test.support import run_command, split_wast
+from scripts.test.shared import *
```
Comment (dschuff) [inline check.py]: So, I'm not a fan of `from foo import *` but I can see the value of doing it this way here.
Can we leave a TODO here describing that we can do more refactoring and/or make the references explicit?

Comment (dschuff) [inline check.py]: Yeah that looks better now.

**Pull request merged**